pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -147,7 +148,15 @@ __maybe_unused static __always_inline void protocol_classifier_entrypoint(struct
         return;
     }
 
-    protocol_stack_t *protocol_stack = get_protocol_stack_if_exists(&usm_ctx->tuple);
+    conn_tuple_t normalized_tup = usm_ctx->tuple;
+    normalize_tuple(&normalized_tup);
```

Comment:
I wonder if the normalization of the tuple should be in the `__get_protocol_stack_if_exists` helper? is there a case where we don't want to normalize it?

Reply:
The purpose of `__get_protocol_stack_if_exists` getting a normalized tuple, is to spare redundant normalization operation on an already normalized stack (happens heavily in the code)
but there's `get_protocol_stack_if_exists` that normalize the conn and then calls `__get_protocol_stack_if_exists`

Reply:
ah that makes sense, sounds good